sdv/metadata/multi_table.py

```diff
@@ -13,6 +15,165 @@ def __init__(self):
         self._tables = {}
         self._relationships = []
 
+    @staticmethod
+    def _validate_missing_relationship_keys(parent_table, parent_table_name, parent_primary_key,
+                                            child_table_name, child_foreign_key):
+        missing_keys = set()
+        parent_primary_key = cast_to_iterable(parent_primary_key)
+        table_primary_keys = set(cast_to_iterable(parent_table._primary_key))
```

Comment:
This chunk of the code (all the way to the error message) can be moved into its own function.

That way you don't have to repeat the same code twice in lines 34-42